<?xml version="1.0"?>

<launch>
    <include file="$(find igvc_navigation)/launch/localization.launch" />

    <!-- Map Node -->
    <node name="mapper" pkg="igvc_navigation" type="mapper" output="screen">
            <param name="topics" type="string" value="/scan/pointcloud"/>
            <param name="occupancy_grid_length" type="int" value="50" />
            <param name="occupancy_grid_width" type="int" value="50" />
            <param name="occupancy_grid_resolution" type="double" value="1" />
            <param name="occupancy_grid_threshold" type="int" value="178" />
            <!-- Set decay_period to 0 to disable decay -->
            <param name="decay_period" type="double" value="60" />
            <param name="transform_max_wait_time" type="double" value="0.5" />
            <param name="start_X" type="double" value="10"/>
            <param name="start_Y" type="double" value="10"/>
            <param name="increment_step" type="int" value="125"/>
            <param name="debug" type="bool" value="false"/>
    </node>

    <arg name="plot_rviz" default="false"/>

    <group if="$(arg plot_rviz)">
        <node name="jessii_rviz" pkg="rviz" type="rviz" args="-d $(find igvc_rviz_plugins)/configs/jessii_conf.rviz"/>
    </group>


    <!-- Planner -->
    <node name="path_planner" pkg="igvc_navigation" type="grid_test" output="screen" >
        <!-- Distance at which a node is considered a valid goal -->
        <param name="goal_range" type="double" value="0.001"/>
        <!-- Robot configuration space -->
        <param name="c_space" type="double" value="0.0"/>
        <!-- Path calculation rate -->
        <param name="rate" type="double" value="1"/>
        <!-- Max distance to a goal node -->
        <param name="maximum_distance" type="double" value="100.0"/>
        <!-- Publish a pointcloud of search nodes on the /expanded topic -->
        <param name="publish_expanded" type="bool" value="true"/>
        <!-- If, for some reason, the path planner decides no optimal path
        currently exists, use the previously generated path instead -->
        <param name="follow_old_path" type="bool" value="true"/>
        <param name="goal_x" type="int" value="10"/>
        <param name="goal_y" type="int" value="40"/>
    </node>
</launch>
